aws/resource_aws_servicecatalog_provisioning_artifact_test.go

```diff
@@ -0,0 +1,457 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	tfservicecatalog "github.com/terraform-providers/terraform-provider-aws/aws/internal/service/servicecatalog"
+)
+
+// add sweeper to delete known test servicecat provisioning artifacts
+func init() {
+	resource.AddTestSweepers("aws_servicecatalog_provisioning_artifact", &resource.Sweeper{
+		Name:         "aws_servicecatalog_provisioning_artifact",
+		Dependencies: []string{},
+		F:            testSweepServiceCatalogProvisioningArtifacts,
+	})
+}
+
+func testSweepServiceCatalogProvisioningArtifacts(region string) error {
+	client, err := sharedClientForRegion(region)
+
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+
+	conn := client.(*AWSClient).scconn
+	sweepResources := make([]*testSweepResource, 0)
+	var errs *multierror.Error
+
+	input := &servicecatalog.SearchProductsAsAdminInput{}
+
+	err = conn.SearchProductsAsAdminPages(input, func(page *servicecatalog.SearchProductsAsAdminOutput, lastPage bool) bool {
+		if page == nil {
+			return !lastPage
+		}
+
+		for _, pvd := range page.ProductViewDetails {
+			if pvd == nil || pvd.ProductViewSummary == nil || pvd.ProductViewSummary.ProductId == nil {
+				continue
+			}
+
+			productID := aws.StringValue(pvd.ProductViewSummary.ProductId)
+
+			artInput := &servicecatalog.ListProvisioningArtifactsInput{
+				ProductId: aws.String(productID),
+			}
+
+			// there's no paginator for ListProvisioningArtifacts
```

Comment:
could also use our generator friend `lister` to make one for us 😄

Reply:
i don't think it will work since there's no token field on the input. you can't even do old-fashioned paging.